iset.mm

```diff
@@ -111968,6 +112091,60 @@ infinite descent (here implemented by strong induction).  (Contributed
   $}
 
 
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Cardinality of real and complex number subsets
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+
+$(
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+  Countability of integers and rationals
+=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
+$)
+
+  ${
+    $d x y z $.
+    $( There are as many odd positive integers as there are positive integers.
+       (Contributed by Jim Kingdon, 11-May-2022.) $)
+    oddennn $p |- { z e. NN | -. 2 || z } ~~ NN $=
+      ( vx vy c2 cv cdvds wbr wn cn c1 co cmin nnex wcel wceq breq2 syl cle a1i
+      cz crab caddc cdiv cmul rabex elrabi peano2nnd elrab simprbi wb oddp1even
+      notbid nnzd mpbid nnehalf syl2anc nnz 2z zmulcld peano2zm 1e2m1 zred nnre
+      remulcld 1red cc0 0le2 nnge1 lemulge12d lesub1dd syl5eqbr elnnz1 sylanbrc
+      dvdsmul2 sylancl oddm1even biimprd mt2d wa eqcom adantr nncnd 1cnd addcld
+      simpr 2cnd cap 2ap0 divmulap3d mulcld addlsub bitrd syl5rbbr en3i ) BCDAE
+      ZFGZHZAIUAZIBEZJUBKZDUCKZCEZDUDKZJLKZWQAIMUEMWSWRNZWTINDWTFGZXAINXEWSWQAW
+      SIUFZUGXEDWSFGZHZXFXEWSINZXIWQXIAWSIWOWSOWPXHWOWSDFPULUHUIXEWSTNXIXFUJXEW
+      SXGUMWSUKQUNWTUOUPXBINZXDINZDXDFGZHZXDWRNXKXDTNZJXDRGXLXKXCTNZXOXKXBDXBUQ
+      ZDTNZXKURSZUSZXCUTQXKJDJLKXDRVAXKDXCJXKDXSVBZXKXBDXBVCZYAVDXKVEXKDXBYAYBV
+      FDRGXKVGSXBVHVIVJVKXDVLVMXKXMDXCFGZXKXBTNXRYCXQURXBDVNVOXKYCHZXMXKXPYDXMU
+      JXTXCVPQVQVRWQXNAXDIWOXDOWPXMWOXDDFPULUHVMXBXAOXAXBOZXEXKVSZWSXDOZXAXBVTY
+      FYEWTXCOYGYFWTXBDYFWSJYFWSXEXJXKXGWAWBZYFWCZWDYFXBXEXKWEWBZYFWFZDVFWGGYFW
+      HSWIYFWSJXCYHYIYFXBDYJYKWJWKWLWMWN $.
+  $}
+
+  ${
+    $d x y z $.
+    $( The Cartesian product of the set of positive integers with itself is
+       equinumerous to the set of positive integers.  (Contributed by NM,
+       1-Aug-2004.)  (Revised by Mario Carneiro, 9-Mar-2013.) $)
+    xpnnen $p |- ( NN X. NN ) ~~ NN $=
+      ( vz vx vy cn c2 cv cdvds wbr wn crab cn0 cxp cexp co cmul cmpt2 wf1o cen
+      ccnv eqid oddpwdc f1ocnv nnex f1oen mp2b oddennn nn0ennn mp2an entr2i
+      xpen ) DEAFGHIADJZKLZDDLZULDBCUKKECFMNBFONPZQDULUNSZQDULRHBCAUNUKUKTUNTUA
+      ULDUNUBDULUOUCUDUEUKDRHKDRHULUMRHAUFUGUKDKDUJUHUI $.
+  $}
```

Comment:
I'd say your are the prime contributor for this version, since your proof is different from Mario's ?

Reply:
Good catch. Assuming that the 2013 revision was to the proof rather than the statement of the theorem, it is a clear-cut case for removing `Revised by Mario Carneiro, 9-Mar-2013`. So I've pushed up a change to do that here and for the similar situation in `xpen`.

As for the `Contributed by` line itself, my usual practice has been to keep it if the statement of the theorem didn't change, or replace it with myself and the current date if the statement of the theorem changed. Although sometimes people have found this rule too simplistic, it is at least an easy one to apply.

My bigger dilemma here was more whether to credit you (@tirix ) since you provided the key idea of using `oddpwdc`. I'll be glad to say in a github comment for sure that that suggestion was very helpful and that I was headed down a rather more difficult path without it, but as for whether/how to reflect that in a `Contributed` line? I guess I rarely have a very clear idea of where to draw any of the lines there.

Reply:
> No problem, my reward is to know that I helped a bit here, and that `oddpwdc` is a useful theorem after all!

Well you are listed as the contributor of `oddpwdc` so perhaps that is the right way to approach this.